pubsub/azuresb/azuresb.go

```diff
@@ -506,16 +525,32 @@ func (s *subscription) updateMessageDispositions(ctx context.Context, ids []driv
 		return nil
 	}
 
-	lockIds := []amqp.UUID{}
-	for _, mid := range ids {
-		if id, ok := mid.(*uuid.UUID); ok {
-			lockTokenBytes := [16]byte(*id)
-			lockIds = append(lockIds, amqp.UUID(lockTokenBytes))
+	// Group by partitionID; AzureSB doesn't support updating dispositions for
+	// messages from different partitions.
+	partitions := map[int16][]amqp.UUID{}
+	for _, ackID := range ids {
+		if pid, ok := ackID.(*partitionAckID); ok {
+			lockTokenBytes := [16]byte(*pid.LockToken)
+			partitions[pid.PartitionID] = append(partitions[pid.PartitionID], amqp.UUID(lockTokenBytes))
 		}
 	}
+
+	// Update all partitions in parallel.
+	g, ctx := errgroup.WithContext(ctx)
+	for _, lockTokens := range partitions {
+		g.Go(func() error {
+			return s.updateMessageDispositionsInPartition(ctx, lockTokens, disposition)
```

Comment:
This will have the behavior that one RPC failing will cancel all the other acks.. is this desirable behavior?

Reply:
No, I guess not; we might as well ack as many as possible. Fixed by no longer using `errgroup`.